### PR TITLE
Added Java17 / record support, added test

### PR DESCRIPTION
**Pull request by blagerweij**

### CODE REVIEW

File: src/main/java/ru/proninyaroslav/template/Exec.java [hunk 1]
```diff
@@ -474,44 +478,44 @@ private Object evalField(final Object dot, final String fieldName, final Node no
...
         boolean hasArgs = args != null && (args.size() > 1 || finalVal != null);
 
-        /* Find methods */
-        final List<Method> foundMethods = new ArrayList<>();
-        for (final Method method : methods) {
-            if (method.getName().equals(fieldName)) {
-                foundMethods.add(method);
+        // first try getter
+        try {
+            PropertyDescriptor pd = new PropertyDescriptor(fieldName, receiver.getClass());
```
Comment (diegourban) [inline src/main/java/ru/proninyaroslav/template/Exec.java]: **suggestion:** add final modifier

File: src/main/java/ru/proninyaroslav/template/Exec.java [hunk 2]
```diff
@@ -474,44 +478,44 @@ private Object evalField(final Object dot, final String fieldName, final Node no
...
+            Method getter = pd.getReadMethod();
```
Comment (diegourban) [inline src/main/java/ru/proninyaroslav/template/Exec.java]: **suggestion:** add final modifier

**Review by diegourban — APPROVED** [review]
lgtm

File: src/main/java/ru/proninyaroslav/template/Exec.java (hunk 2 above)
Comment (blagerweij) [reply src/main/java/ru/proninyaroslav/template/Exec.java]: Done

**Pull request merged**